x/dex/keeper/withdraw_filled_limit_order.go

```diff
@@ -98,7 +98,7 @@ func (k Keeper) ExecuteWithdrawFilledLimitOrder(
 
 	}
 
-	k.SaveTrancheUser(ctx, trancheUser)
+	k.SaveOrRemoveTrancheUser(ctx, trancheUser)
```

Comment:
we're withdrawing a filled limit order here. can it be the case that we go to the second branch of the `SaveOrRemoveTrancheUser` that does `SetLimitOrderTrancheUser` in case if `!trancheUser.IsEmpty()`?

Reply:
also could you please have a look at all the combined methods and

1) check whether they are applied correctly or there can only be one of the conditions met;
2) if combination is applied correctly in several places, which would imply that we need that method to avoid code duplication, cover the method and its usage with comments so it's clear for the reader how it works and why its called there